### PR TITLE
Update serenity and clean up commands and event handlers in main

### PR DESCRIPTION
**Pull request by technetos**
This PR adds no new functionality, its just code cleanup and some updates

~hmm but it seems i broke the welcome message....~

### CODE REVIEW

File: src/commands.rs [hunk 1]
```diff
@@ -205,149 +198,139 @@ impl Commands {
...
-fn add_code_segment_multi_line<T>(
-    state_machine: &mut StateMachine<T>,
-    name: &'static str,
-    mut state: usize,
-) -> usize {
-    state = state_machine.add(state, CharacterSet::from_char('`'));
-    state = state_machine.add(state, CharacterSet::from_char('`'));
-    state = state_machine.add(state, CharacterSet::from_char('`'));
+    fn add_remaining_segment(&mut self, mut state: usize, s: &'static str) -> usize {
+        let name = &s[..s.len() - 3];
```
Comment (khionu) [inline src/commands.rs:239]: Is this guaranteed to operate on character boundaries? Indexes are on bytes.

**Review by khionu — CHANGES_REQUESTED** [review]
Only concern being the string indexing. Can't take enough of a look atm to ensure that it won't try to pull data mid character.

File: src/commands.rs (hunk 1 above)
Comment (technetos) [reply src/commands.rs:239]: Yeah its fine, the only characters that fit in those states are ascii.

Comment (technetos) [reply src/commands.rs:239]: We have been doing this indexing all along, i just moved it

**Review by khionu — APPROVED** [review]

**Pull request merged**